plenum/test/view_change/test_different_last_ordered_before_view_change.py

```diff
@@ -0,0 +1,154 @@
+                           2,
+                           slow_instance))
+
+            # trigger view change on all nodes
+            for node in txnPoolNodeSet:
+                node.view_changer.on_master_degradation()
+
+            # wait for view change done on all nodes
+            looper.run(eventually(view_change_done, txnPoolNodeSet, old_view_no + 1))
+
+    last_ordered_3pc = fast_nodes[0].replicas[slow_instance].last_ordered_3pc
+    for node in txnPoolNodeSet:
+        assert last_ordered_3pc == node.replicas[slow_instance].last_ordered_3pc
+    sdk_send_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle,
+                              sdk_wallet_client, 1)
+    looper.run(
+        eventually(last_ordered,
+                   txnPoolNodeSet,
+                   (txnPoolNodeSet[0].viewNo, 1),
+                   slow_instance))
+
+
+def test_different_last_ordered_on_master_before_view_change(looper, txnPoolNodeSet,
+                                                  sdk_pool_handle, sdk_wallet_client):
+    ''' Send random request and do view change then fast_nodes (1, 4 - without
+    primary after next view change) are already ordered transaction on master
+    and slow_nodes are not. Check ordering on slow_nodes.'''
+    sdk_send_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle,
+                              sdk_wallet_client, 1)
+    master_instance = txnPoolNodeSet[0].master_replica.instId
+    slow_nodes = txnPoolNodeSet[1:3]
+    fast_nodes = [n for n in txnPoolNodeSet if n not in slow_nodes]
+    nodes_stashers = [n.nodeIbStasher for n in slow_nodes]
+    old_last_ordered = txnPoolNodeSet[0].master_replica.last_ordered_3pc
+    with delay_rules(nodes_stashers, cDelay(delay=sys.maxsize)):
+        # send one  request
+        requests = sdk_send_random_requests(looper, sdk_pool_handle,
+                                             sdk_wallet_client, 1)
+        last_ordered_for_slow = slow_nodes[0].master_replica.last_ordered_3pc
+        old_view_no = txnPoolNodeSet[0].viewNo
+        looper.run(
+            eventually(last_ordered,
+                       fast_nodes,
+                       (old_view_no, old_last_ordered[1] + 1),
+                       master_instance))
+
+        # trigger view change on all nodes
+        for node in txnPoolNodeSet:
+            node.view_changer.on_master_degradation()
+
+        # wait for view change done on all nodes
+        looper.run(eventually(view_change_done, txnPoolNodeSet, old_view_no + 1))
+
+    sdk_get_replies(looper, requests)
+    looper.run(
+        eventually(last_ordered,
+                   slow_nodes,
+                   (old_view_no, last_ordered_for_slow[1] + 1),
+                   master_instance))
+
+
+def last_ordered(nodes: [Node],
+                 last_ordered,
+                 instId):
+    for node in nodes:
+        assert node.replicas[instId].last_ordered_3pc == last_ordered
+
+
+def view_change_done(nodes: [Node], view_no):
```

Comment:
This doesn't indicate end of view change, rather the beginning.
Please call `ensureElectionsDone` in addition to make sure that it's finished.

Reply:
Done.